### PR TITLE
CI/CD pipeline

### PR DESCRIPTION
**Pull request by syedhassaanahmed**
Based on #10 I've added a CI/CD pipeline in Travis. 

- Small changes in `build.sbt`. `version` and `scalaVersion` are now parametrized with default values, so that we can control them from Travis.
- Build is executed against multiple Scala versions (2.11 and 2.12), however GitHub release will only be uploaded for v2.11.
- [GitHub release uploading](https://docs.travis-ci.com/user/deployment/releases/) is enabled and **encrypted** OAuth token was generated using Travis CLI; 
```travis setup releases --force```
All you have to do is login to Travis with your GitHub account, enable CI on this repo, create your encrypted token with Travis CLI and replace it in `.travis.yml`. (**Discard all changes other than the token!!!**)
- The deployment is triggered only when a tag is pushed.
- If a build fails, Travis user will be notified via email.
- Build status is shown in README.

---
This change is [Reviewable](https://reviewable.io/reviews/azure/toketi-kafka-connect-iothub/17)

### CODE REVIEW

**Comment by varunpuranik** [comment]
Hi Syed, 

Thanks for helping set up the CI/CD pipeline for this repo. I will merge your PR and set it up in travis soon. 

Thanks,
-Varun

**Comment by syedhassaanahmed** [comment]
Hi @varunpuranik 
Any update on this?

**Comment by varunpuranik** [comment]
Sorry about the delay, I haven't had a chance to get to this. I will try to get to it by this weekend. 

Thanks,
-Varun

File: build.sbt [hunk 1]
```diff
@@ -1,12 +1,12 @@
 // Copyright (c) Microsoft. All rights reserved.
 
-val iotHubKafkaConnectVersion = "0.7.0"
+val iotHubKafkaConnectVersion = util.Properties.propOrNone("version").getOrElse("0.7.1")
```
Comment (dluc) [inline build.sbt:3]: what does `util.Properties.propOrNone("version")` do? e.g. where is the value retrieved from?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+language: scala
+scala:
+- 2.11.11
+- 2.12.2
+script:
+- sbt -DscalaVersion=$TRAVIS_SCALA_VERSION assembly
+deploy:
+  provider: releases
+  api_key:
+    secure: pH3qOqoZQYf6G7HMG9gPfaLGxElyi0HN0jlee+mdqGKU/xfULjNIah0gp3JlHdDqkfm+jOBMxjjICqtEwXtLnyd9YJ+W/ZyzS4hhIJadhh3Mcb/8UvZ+dAw3bsCQ70HAjfevwTPaE/kbbLuoeKWfO3MLjjEYX3Q4KVMegUw7PjSy2KleVI6t9KDYwqmTqpaes66k8U9hQx6mGBbwt761G9V8UOb9Eo1OqhG0OaoEjlo+5fd3e7fI41i4pRNAMgGBe27pjKBywUFZPHe089/7SyWxfrbdHei8iU3IEBVA8tAtzydHnq2C1rj7x42KkIkn7tH/g/zZYCTai0BwomWKLzaDf3x1ZueMpReCPpH/ILHiiDkfqzfmQ+YfQvDbvmsoWIXRCiLZeB9sc6ccEsK53q/Mqa2fdICUFtDeS7yUwovC0NZqJvnGgr2g20y/StR48XhScMjKDWOsHYmMcXQ/Fb0n9ZZPuay0t48WXhI5fqx9GJ6jY6FZJ1SpDwuZRQYaVKXr+Y9kHnMXWXMd2pJgew4XWZQW9XFdZsYBGGCb5eIpEnW3wmbPimCWKWfw3Mk5BNUnO9HsNmvOS01+isvX5eMHTxwIdH/CuFy75IocnA8of2ju4HG4cQaRMnDLc5n5/eUOTd+41ig13hEe9cv+jFTGhvk6BbQ6bGF6tvgr5Zk=
```
Comment (dluc) [inline .travis.yml]: I assume this is a secret pointing to resources that you own. Could you provide some more info? Is it possible to replace this with environment variables instead?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+language: scala
```
Comment (dluc) [inline .travis.yml:2]: this is a Travis conf I use in another project, what do you think? we could cache the folders to speed up the build
```
jdk: oraclejdk8
language: scala
scala:
- 2.11.11
- 2.12.2
cache:
  directories:
  - "$HOME/.ivy2"
  - "$HOME/.sbt"
  - "$HOME/.m2"
```

File: build.sbt (hunk 1 above)
Comment (syedhassaanahmed) [reply build.sbt:3]: @dluc `util.Properties.propOrNone("version")` allows us to read params passed to the `sbt tool`. e.g. `sbt -Dversion=0.7.1`. Travis could pass this param in the `script` section when it calls sbt (currently it only passes scalaVersion).

File: .travis.yml (hunk 2 above)
Comment (syedhassaanahmed) [reply .travis.yml]: Its the GitHub deployment OAuth token currently pointing to my fork (should be replaced with yours as mentioned in PR description). Even though its encrypted, I agree that its a better practice to  replace it with an environment variable instead. Change done.

File: .travis.yml (hunk 3 above)
Comment (syedhassaanahmed) [reply .travis.yml:2]: @dluc Makes sense. I've added caching as well.

**Comment by syedhassaanahmed** [comment]
Closing the PR as the project doesn't seem to be active.

**Pull request closed without merging**